File: app/src/main/java/com/djfos/im/filter/FilterGrayScale.java
package com.djfos.im.filter;


import org.opencv.core.Mat;
import org.opencv.imgproc.Imgproc;

public class FilterGrayScale extends BaseFilter {
    FilterGrayScale() {
        type = FilterType.GARY_SCALE;
    }

    @Override
    Mat apply(Mat in) {
        Mat mat = new Mat();
        Imgproc.cvtColor(in, mat, Imgproc.COLOR_RGBA2GRAY);
        return mat;
    }
}

File: app/src/main/java/com/djfos/im/filter/FilterGrayScale.kt
package com.djfos.im.filter


import org.opencv.core.Mat
import org.opencv.imgproc.Imgproc


class FilterGrayScale : AbstractFilter() {
    override val type = FilterType.Gray
    override fun apply(input: Mat): Mat {
        val mat = Mat()
        Imgproc.cvtColor(input, mat, Imgproc.COLOR_RGBA2GRAY)
        return mat
    }
}


